sycl/test-e2e/Basic/cospi.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <iomanip>
+#include <iostream>
+#include <sycl.hpp>
+
+template <typename T> T get_ulp_std(T x) {
+  const T inf = std::numeric_limits<T>::infinity();
+  const T negative = std::fabs(std::nextafter(x, -inf) - x);
+  const T positive = std::fabs(std::nextafter(x, inf) - x);
+  return std::fmin(negative, positive);
+}
+
+int main() {
+  const double value = 0.4863334355; // some random value
+  const double reference =
+      0.042921588887841428949787569990803604014217853546142578125; // calculated
+                                                                   // with
+                                                                   // oclmath
+  double res;
+
+  {
+    sycl::queue queue;
+    sycl::buffer<double, 1> cospi_buf(&res, 1);
+    queue
+        .submit([&](sycl::handler &cgh) {
+          auto cospi_acc = cospi_buf.get_access(cgh);
+          cgh.single_task([=]() { cospi_acc[0] = sycl::cospi(value); });
+        })
+        .wait_and_throw();
+  }
+
+  const unsigned int ulpsExpected = 4; // from OpenCL spec
```

Comment:
We should not rely on OpenCL spec in SYCL tests. We have other backends which can have different accuracy.

Reply:
Can you please extend the comment in this case? It could be like this, feel free to rephrase 

```
// according to section 4.17.5 of SYCL 2020 spec, math functions correspond to math functions from OpenCL 1.2 spec, so getting the accuracy here from OpenCL 1.2 spec
```